tutorial/README.md

```diff
@@ -0,0 +1,824 @@
+    // name should be the the same as the component name.
+    workload: {
+      apiVersion: 'core.oam.dev/v1alpha2'
+      kind: 'ContainerizedWorkload'
+      metadata: {
+        name: 'hello-world'
+      }
+
+      // The spec section (inside the workload) contains the properties that configure the container.
+      spec: {
+
+        // Our workloads are all running on linux-amd64 but other platforms can be supported.
+        osType: 'linux'
+        arch: 'amd64'
+
+        // A list of containers to run and their relevant settings. This should be familiar
+        // if you've ever used docker-compose or Kubernetes.
+        containers: [
+          {
+            name: 'hello-world'
+            image: 'nginxdemos/hello'
+            ports: [
+              {
+                name: 'http'
+                containerPort: 80
+                protocol: 'TCP'
+              }
+            ]
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+ARM-DSL is a JSON-like language intended for tersely specifying objects. Arrays are defined with `[]`, objects are defined with `{}`, strings are *single-quoted* (`'this is a string'`).
+
+Defining a `Component` is like specifying a *template for a container* - nothing runs as a result of defining a `Component`. What's needed to deploy the application is to define an `ApplicationConfiguration`. 
+
+The `ApplicationConfiguration` used in this step has its contents displayed here for learning purposes. You can find it on disk at `./hello-world/applicationconfiguration.arm`.
+
+```js
+// Like a Component, the ApplicationConfiguration also uses a resource statement to declare an object.
+resource oam 'ApplicationConfiguration@core.oam.dev/v1alpha2' _ {
+
+  // Like a Component, the ApplicationConfiguration also must declare a metadata section and a name.
+  metadata: {
+    name: 'hello-world'
+  }
+
+  // The spec section contains the details of the ApplicationConfiguration. 
+  // The primary thing that it should contain is a list of Components.
+  spec : {
+    components: [
+      {
+        // Refers to the name of the Component declared above.
+        componentName: 'hello-world'
+      }
+    ]
+  }
+}
+```
+
+Applying an `ApplicationConfiguration` as well as all of the `Component` definitions it contains will deploy the application to Kubernetes.
+
+---
+
+Now it's time to deploy this example. Make sure you have an active terminal for both the control plane VM (running minikube) and the AKS cluster. You will use the control plane VM to deploy manifests and the applications will run in the AKS cluster.
+
+Your control plane virtual machine wil already have `azoam` and `kubectl` on the path so they can be run. 
+
+Using the control plane vm:
+
+```sh
+# you should be logged in to the control plane VM as 'azureuser'
+whoami
+
+# minikube should be running - use `minikube start` if it is not
+minikube start
+
+# deploy the application
+azoam hello-world/* | kubectl apply -f -
```

Comment:
This is not in the VM at all.

Reply:
That's pretty strange, it was when I did it. I'll delete my setup and check.

Reply:
This definitely worked for me. If you were using the `_templateArtifacts` parameter (remember there was a mistake where it was present in the instructions) the you might have used an older copy of the script. I just ran through this again using the instructions in master, and I have `azoam`

```
azureuser@oam-vm-u2vzbhk5tynfs:~/azure-oam-solution$ type azoam
azoam is hashed (/usr/local/bin/azoam)
```

Reply:
I am referring to the hello-world folder.

Reply:
The instructions say to clone the repo. That's definitely part of the expectations.